### ISSUE TITLE
Generating SLSA provenance and SBOM 

### ISSUE THREAD

**Issue by bpradipt** [confidential-containers/confidential-containers#219]
@mkulke @Xynnn007 @zvonkok  and I got together to list the current state of provenance management in CoCo, gaps and next steps.

A Google doc describing the current PoC, including open questions, is available here - https://docs.google.com/document/d/1HmeY9cJ0VKGg1U2EQrhJCloMiufF1LE9XEFbpzW7ZqM/edit

Our immediate next step is to introduce support for generating SLSA provenance and SBOM for CoCo artefacts. 

Initially, the idea is to start with a few components (e.g. guest components)  and then replicate it for other projects.

This issue is a placeholder for requesting feedback and comments from the community.

**Comment by Xynnn007** [comment]
Yes, I think AA/CDH/ASR/KBS-client would be a good start to apply SLSA provenance things.

It would give a detailed metadata for the building process for the binaries.

@fitzthum might be interested also.

**Comment by mkulke** [comment]
In the scope of this [GC PR](https://github.com/confidential-containers/guest-components/pull/731) we started pushing SLSA provenance reports to guest-components' OCI registry. Peerpods intends to verify those attestations as it consumes the GC binaries.

The respective Github feature has been make[ generally available](https://docs.github.com/en/actions/security-for-github-actions/using-artifact-attestations/using-artifact-attestations-to-establish-provenance-for-builds).

**Comment by mkulke** [comment]
we also added it to [kata-agent](https://github.com/kata-containers/kata-containers/blob/4c34cfb0abf982208b39ec585cba635406ecff0e/.github/workflows/build-kata-static-tarball-amd64.yaml#L71)
